src/code/utils/topology-tagger.ts

```diff
@@ -78,6 +78,13 @@ function isLinearModel(g) {
     .reduce ( (a, b) => (a + b), 0) === g.nodeCount();
 }
 
+function anyBranches(g) {
+  const brancedOrJoinedNodeCount = g.nodes()
+    .map( (node) => ((g.inEdges(node).length > 1 || g.outEdges(node).length > 1 ) ? 1 : 0))
```

Comment:
There is an opportunity to bail early if we use `Array.find()` instead of `Array.map()`

But it's probably an unimportant optimization at this point.